app/models/group.rb

```diff
@@ -168,6 +168,7 @@ class Group < ActiveRecord::Base # rubocop:disable Metrics/ClassLength
 
   validates_by_schema except: [:logo]
   validates :type, uniqueness: { scope: :parent_id }, if: :static_name
+  validates :parent, presence: true, unless: -> { Group.roots.count.zero? || id == Group.root.id }
```

Comment:
This should work in the following edge cases I thought off:
 * Creating the first group: `Group.roots.count.zero?` will be true
 * The root group will still be valid when it has been created since `id == Group.root.id` has been created.
 * Further groups must have a `parent` present.